src/weekday.rs

```diff
@@ -50,14 +50,29 @@ pub enum Weekday {
 }
 
 impl Weekday {
+    /// Create a `Weekday` from an `u32`, with Monday = 0.
+    /// Infallible, takes `n & 7`.
+    #[inline]
+    pub(crate) const fn from_u32_mod7(n: u32) -> Weekday {
```

Comment:
For use in this crate it would be nice if there was a const, infallible way to create a `Weekday`.

Because the two methods construction a `Weekday` in `internals.rs` passed `n % 7`, making `from_u32_mod7` seemed like a good way to have an infallible function and avoid a little code duplication.

Reply:
Given the `mod7` requirement which is not enforced in the type system and the fact that this is used only in the `internals` module, I think this should be a free function in the `internals` module rather than a method here.